src/Integration/Suppression/SuppressedIssuesFileStorage.cs

```diff
@@ -0,0 +1,113 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.Suppression
+{
+    internal class SuppressedIssuesFileStorage : ISuppressedIssuesFileStorage
+    {
+        private readonly string fileDirectory;
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+
+        public SuppressedIssuesFileStorage(ILogger logger) : this(logger, new FileSystem())
+        {
+            
+        }
+
+        internal SuppressedIssuesFileStorage(ILogger logger, IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+
+            fileDirectory = Path.Combine(Path.GetTempPath(), "SLVS", "Roslyn");
+        }
+
+        public IEnumerable<SonarQubeIssue> Get(string sonarProjectKey)
+        {
+            try
+            {
+                if (!ValidateFileName(sonarProjectKey))
+                {
+                    return Enumerable.Empty<SonarQubeIssue>();
+                }
+
+                var filePath = GetFilePath(sonarProjectKey);
+                var fileContent = fileSystem.File.ReadAllText(filePath);
+                return JsonConvert.DeserializeObject<IEnumerable<SonarQubeIssue>>(fileContent);
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLine($"SuppressedIssuesFileStorage Get Error: {ex.Message}");
```

Comment:
In general all logged messages except "debug"-level messages should be localised i.e. the text should be in a resource file.